File: my-folder/1421-find-numbers-with-even-number-of-digits/solution.c
int get_digit_count(int val){
    int result = 0;
    while (val != 0){
        result++;
        val /= 10;
    }
    return result;
}

int findNumbers(int* nums, int numsSize) {
    int result = 0;
    for (size_t i = 0; i < numsSize; i++){
        result += (get_digit_count(nums[i]) % 2 == 0 ? 1 : 0);
    }
    return result;
}

File: my-folder/1421-find-numbers-with-even-number-of-digits/solution.rs
impl Solution {
    pub fn find_numbers(nums: Vec<i32>) -> i32 {
        nums.iter()
            .map(|num| (*num as f64).log10())
            .filter(|digits| (*digits as u32) % 2 != 0)
            .count() as i32
    }
}
